src/main/java/de/tum/in/www1/artemis/web/rest/GradingInstructionResource.java

```diff
@@ -0,0 +1,151 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.forbidden;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.GradingInstruction;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.ExerciseService;
+import de.tum.in.www1.artemis.service.GradingInstructionService;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+/**
+ * REST controller for managing Structured Grading Instructions.
+ */
+@RestController
+@RequestMapping({ "/api", "/api_basic" })
+@PreAuthorize("hasAnyRole('INSTRUCTOR','ADMIN')")
+public class GradingInstructionResource {
+
+    private final Logger log = LoggerFactory.getLogger(GradingInstructionResource.class);
+
+    private static final String ENTITY_NAME = "gradingInstruction";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final GradingInstructionService gradingInstructionService;
+
+    private final ExerciseService exerciseService;
+
+    private final AuthorizationCheckService authCheckService;
+
+    public GradingInstructionResource(GradingInstructionService gradingInstructionService, ExerciseService exerciseService, AuthorizationCheckService authCheckService) {
+        this.gradingInstructionService = gradingInstructionService;
+        this.exerciseService = exerciseService;
+        this.authCheckService = authCheckService;
+    }
+
+    /**
+     * GET /exercises/:id : get the "id" exercise.
+     *
+     * @param exerciseId the id of the exercise to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise, or with status 404 (Not Found)
+     */
+    @GetMapping("/grading-instruction/{exerciseId}")
```

Comment:
The route name is not conform with our REST naming conventions (see [here](https://confluence.ase.in.tum.de/display/ArTEMiS/Server)). Specifically, it violates these points:

1. plural of entity name should normally be used (`/grading-instructions`)
2. the placement of the `exerciseId` param is misleading since a call to e.g. `/grading-instruction/123` insinuates that `123` is the gradient instruction id (and not the exercise id).
3. the url structure should go from entity to subentity, meaning in this case `/exercises/{exerciseId}/gradient-instructions` would probably be a good choice